notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -47,6 +48,31 @@ class JiraNotifier(
         restClient.issueClient.addComment(issue.commentsUri, Comment.valueOf(comment)).claim()
     }
 
+    /**
+     * Change the [state] of the given issue specified by the [issueKey].
+     */
+    fun changeState(issueKey: String, state: String): Result<Boolean> {
+        val issue = restClient.issueClient.getIssue(issueKey).claim()
+        val possibleTransitions = restClient.issueClient.getTransitions(issue).claim()
+
+        possibleTransitions.forEach { transition ->
+            if (state == transition.name) {
+                return try {
+                    restClient.issueClient.transition(issue, TransitionInput(transition.id)).claim()
+
+                    Result.success(true)
+                } catch (e: RestClientException) {
+                    log.error { "The transition to state '$state' for the issue '$issueKey' is not possible: " +
+                            e.collectMessagesAsString() }
+
+                    Result.failure(e)
+                }
```

Comment:
You should be able to slightly simplify this, similar to like I just did in https://github.com/oss-review-toolkit/ort/pull/4944/commits/85fcab0c62080fb1858008962202fb4a3091fe93.

Reply:
Ah nice! Due to the changes from https://github.com/oss-review-toolkit/ort/commit/f24fa74edb96056e2d7199016ca32cadb8df6829 we can use `runCatching` now 👍